gateway/src/shard/processor/session.rs

```diff
@@ -59,10 +59,10 @@ impl Error for SessionSendError {
 pub struct Session {
     // Needs to be Arc so it can be cloned in the `Drop` impl when spawned on
     // the runtime.
-    pub heartbeater_handle: Arc<Mutex<Option<AbortHandle>>>,
+    pub heartbeater_handle: Arc<MutexSync<Option<AbortHandle>>>,
     pub heartbeats: Arc<Heartbeats>,
     pub heartbeat_interval: AtomicU64,
-    pub id: Mutex<Option<String>>,
```

Comment:
Should we bite the bullet and import parking lot here?

Reply:
`parking_lot` is currently only brought in if `metrics` is enabled. In my opinion I don't think there's too much benefit. Sending commands, retrieving shards, and getting shard info aren't operations that are done that much, so I don't see a lot of - if any - contention. I don't see for a reason for these locks to poison